modules/ollama/local.go

```diff
@@ -0,0 +1,724 @@
+// It returns proto://host:port string for the given exposed port.
+// It returns just host:port if proto is blank.
+func (c *localProcess) PortEndpoint(ctx context.Context, port nat.Port, proto string) (string, error) {
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", fmt.Errorf("host: %w", err)
+	}
+
+	outerPort, err := c.MappedPort(ctx, port)
+	if err != nil {
+		return "", fmt.Errorf("mapped port: %w", err)
+	}
+
+	if proto != "" {
+		proto += "://"
+	}
+
+	return fmt.Sprintf("%s%s:%s", proto, host, outerPort.Port()), nil
+}
+
+// SessionID implements testcontainers.Container interface for the local Ollama binary.
+func (c *localProcess) SessionID() string {
+	return c.sessionID
+}
+
+// Deprecated: it will be removed in the next major release.
+// FollowOutput is not implemented for the local Ollama binary.
+// It panics if called.
+func (c *localProcess) FollowOutput(consumer testcontainers.LogConsumer) {
+	panic("not implemented")
+}
+
+// Deprecated: use c.Inspect(ctx).NetworkSettings.Ports instead.
+// Ports gets the exposed ports for the container.
+func (c *localProcess) Ports(ctx context.Context) (nat.PortMap, error) {
+	inspect, err := c.Inspect(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return inspect.NetworkSettings.Ports, nil
+}
+
+// Deprecated: it will be removed in the next major release.
+// StartLogProducer implements testcontainers.Container interface for the local Ollama binary.
+// It returns an error because the local Ollama binary doesn't have a log producer.
+func (c *localProcess) StartLogProducer(context.Context, ...testcontainers.LogProductionOption) error {
+	return errors.ErrUnsupported
+}
+
+// Deprecated: it will be removed in the next major release.
+// StopLogProducer implements testcontainers.Container interface for the local Ollama binary.
+// It returns an error because the local Ollama binary doesn't have a log producer.
+func (c *localProcess) StopLogProducer() error {
+	return errors.ErrUnsupported
+}
+
+// Deprecated: Use c.Inspect(ctx).Name instead.
+// Name returns the name for the local Ollama binary.
+func (c *localProcess) Name(context.Context) (string, error) {
+	return localNamePrefix + "-" + c.sessionID, nil
+}
+
+// Customize implements the [testcontainers.ContainerCustomizer] interface.
+// It configures the environment variables set by [WithUseLocal] and sets up
+// the wait strategy to extract the host, port and version from the log.
+func (c *localProcess) Customize(req *testcontainers.GenericContainerRequest) error {
+	// Replace the default host port strategy with one that waits for a log entry
+	// and extracts the host, port and version from it.
+	if err := wait.Walk(&req.WaitingFor, func(w wait.Strategy) error {
+		if _, ok := w.(*wait.HostPortStrategy); ok {
+			return wait.VisitRemove
+		}
+
+		return nil
+	}); err != nil {
+		return fmt.Errorf("walk strategies: %w", err)
+	}
+
+	logStrategy := wait.ForLog(localLogRegex).Submatch(c.extractLogDetails)
```

Comment:
Ok, now I see why you needed to add here the changes in the wait strategy. I was about to ask for a different PR for that, but it's fine for me to merge it here.

Reply:
My intent is to split that into a separate PR, if this looks good.

Reply:
Give me a thumbs up @mdelapenya if you like the approach and I'll create the separate wait PR with docs, which we can merge first.

Reply:
Went ahead and did the split you can find it [here](https://github.com/testcontainers/testcontainers-go/pull/2929), just waiting on tests to pass to take it out of draft.